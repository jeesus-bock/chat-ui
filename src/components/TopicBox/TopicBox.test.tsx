import { describe, expect, test } from 'vitest';
import { render } from 'solid-testing-library';
import { TopicBox } from './index';

describe('<TopixBox />', () => {
  test('renders', () => {
    const { container, unmount } = render(() => <TopicBox room='#test_room' topic='Testing topic' />);
    expect(container).toMatchSnapshot();
    unmount();
  });

  test('presents message data', async () => {
    const { queryByTestId } = render(() => <TopicBox room='#test_room' topic='Testing topic' />);
    expect(queryByTestId('topic')).toHaveTextContent('#test_room - Testing topic');
  });
});
