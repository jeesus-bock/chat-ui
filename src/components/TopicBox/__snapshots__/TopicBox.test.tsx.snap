// Vitest Snapshot v1

exports[`<TopixBox /> > renders 1`] = `
<div>
  <div
    class="w-full p-4 border-b border-gray-500 bg-stone-100"
  >
    <label
      data-testid="topic"
    >
      <!--#-->
      #test_room
      <!--/-->
       - 
      <!--#-->
      Testing topic
      <!--/-->
    </label>
  </div>
</div>
`;
